src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerCorporateValidator.java

```diff
@@ -52,6 +54,15 @@ public Errors validate(List<ManagingOfficerCorporateDto> managingOfficerCorporat
             validateRoleAndResponsibilities(managingOfficerCorporateDto.getRoleAndResponsibilities(), errors, loggingContext);
             validateContactFullName(managingOfficerCorporateDto.getContactFullName(), errors, loggingContext);
             validateContactEmail(managingOfficerCorporateDto.getContactEmail(), errors, loggingContext);
+
+            if (managingOfficerCorporateDto.getStartDate() != null) {
+                validateStartDate(managingOfficerCorporateDto.getStartDate(), errors, loggingContext);
+                validateResignedOnDateAgainstStartDate(managingOfficerCorporateDto.getResignedOn(), managingOfficerCorporateDto.getStartDate(), errors, loggingContext);
```

Comment:
there is no null check performed on getResignedOn() so the resigned_on date could potentially be null on line 60. the validateResignedOnDateAgainstStartDate method will pass the resigned_on date into DateValidators.isDateInPast, which will try and call .isAfter() on the null resigned_on (if null at this point) causing a null pointer exception. You'd need an extra null check against resigned_on before comparing against the start_date to make sure both dates are not null.

Reply:
Good spot and my bad (as I had the null check in the else).